### ISSUE TITLE
Incorrect diagnostic wording for out-of-line enum declarations

### ISSUE THREAD

**Issue by Endilll** [llvm/llvm-project#134955]
Consider the following (https://godbolt.org/z/cj1hMqbof):
```cpp
struct A {
  enum E : int;
};

namespace N {
enum A::E : int { e1 };
}
```
Clang rejects this with `error: cannot define or redeclare 'E' here because namespace 'N' does not enclose namespace 'A'`, but `A` is a struct and not a namespace.

**Comment by zygoloid** [comment]
> Clang rejects this with `error: cannot define or redeclare 'E' here because namespace 'N' does not enclose namespace 'A'`, but `A` is a struct and not a namespace.

Perhaps removing the word "namespace" would help?

> Edit: more importantly, it seems that this code is well-formed per https://eel.is/c++draft/enum#dcl.enum-4:

That paragraph only contains restrictions, so can't make code well-formed. It looks to me like this example violates [[expr.prim.id.qual]/2](https://eel.is/c++draft/expr.prim.id.qual#2.sentence-3):

> A declaration that uses a declarative nested-name-specifier shall be a friend declaration or inhabit a scope that contains the entity being redeclared or specialized.

The scope of namespace `N` does not contain `A::E`.

**Comment by Endilll** [comment]
> It looks to me like this example violates [[expr.prim.id.qual]/2](https://eel.is/c++draft/expr.prim.id.qual#2.sentence-3):

Oh, I missed that wording. I agree that the example is ill-formed.
